CMakeLists.txt

```diff
@@ -114,13 +118,43 @@ endif()
 include(${BLT_SOURCE_DIR}/SetupBLT.cmake)
 
 #------------------------------------------------------------------------------
-# Setup Macros and dependencies
+# Setup Macros/Basics
 #------------------------------------------------------------------------------
 
 include(${PROJECT_SOURCE_DIR}/cmake/SeracMacros.cmake)
 include(${PROJECT_SOURCE_DIR}/cmake/SeracBasics.cmake)
 include(${PROJECT_SOURCE_DIR}/cmake/SeracCompilerFlags.cmake)
 
+#------------------------------------------------------------------------------
+# Add Code Checks
+#------------------------------------------------------------------------------
+
+message(STATUS "Serac Code Checks: ${SERAC_ENABLE_CODE_CHECKS}")
+
+# Add extra file extensions for code styling
+# Note: Add them also to serac_add_code_checks in cmake/SeracMacros.cmake
+
+# CUDA
+list(APPEND BLT_C_FILE_EXTS ".cuh")
+# Configured C++ files
+list(APPEND BLT_C_FILE_EXTS ".cpp.in" ".hpp.in")
+
+if (SERAC_STYLE_ONLY OR SERAC_ENABLE_CODE_CHECKS)
+    serac_add_code_checks(PREFIX serac)
+endif()
+
+if (SERAC_STYLE_ONLY)
```

Comment:
Already approved, but maybe should be renamed to `SERAC_CI_STYLE_ONLY`? To not confuse users in case they see the variable.